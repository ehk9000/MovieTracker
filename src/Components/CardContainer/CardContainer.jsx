import React from 'react';
import Card from '../Card/Card';
import { connect } from 'react-redux';
import Styles from './CardContainer.scss';
import PropTypes from 'prop-types';

const CardContainer = props => {
	const nowPlaying = props.nowPlaying.nowPlaying;
	const popular = props.popular.popular;

	let display;

	if (props.category === 'nowPlaying' && nowPlaying !== undefined) {
		display = nowPlaying.map(movie => {
			return <Card id={movie.id} title={movie.title} image={movie.poster_path} />;
		});
	}

	if (props.category === 'popular' && popular !== undefined) {
		display = popular.map(movie => {
			return <Card id={movie.id} title={movie.title} image={movie.poster_path} />;
		});
	}

	return <main className="cardContainer">{display}</main>;
};

const mapStateToProps = ({ popular, nowPlaying }) => ({
	popular: popular,
	nowPlaying: nowPlaying
});

export default connect(mapStateToProps)(CardContainer);

CardContainer.propTypes = {
	popular: PropTypes.array,
	nowPlaying: PropTypes.array,
	id: PropTypes.number,
	image: PropTypes.string,
	title: PropTypes.string
};
