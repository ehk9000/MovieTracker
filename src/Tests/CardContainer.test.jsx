import React from 'react';
import ReactDOM from 'react-dom';
import {shallow} from 'enzyme';
import {CardContainer} from '../Components/CardContainer/CardContainer';


describe('CardContainer', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow( <CardContainer/> );
  });

  it('should mock snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

})