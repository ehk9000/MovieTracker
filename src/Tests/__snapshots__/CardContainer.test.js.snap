// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should mock snapshot 1`] = `undefined`;

exports[`CardContainer should mock snapshot 1`] = `undefined`;
